// @republic-global/payment-client/dist/src/schemas.gen.js
// This file is auto-generated by @hey-api/openapi-ts
export const $CreatePaymentRequest = {
    type: 'object',
    properties: {
        userId: {
            type: 'string',
        },
        amount: {
            type: 'integer',
            format: 'int32',
        },
        description: {
            type: 'string',
        },
        externalId: {
            type: 'string',
        },
    },
};
export const $Payment = {
    type: 'object',
    properties: {
        id: {
            type: 'integer',
            format: 'int64',
        },
        userId: {
            type: 'string',
        },
        amount: {
            type: 'integer',
            format: 'int32',
        },
        description: {
            type: 'string',
        },
        status: {
            type: 'string',
        },
        originalAmount: {
            type: 'integer',
            format: 'int32',
        },
        createdAt: {
            type: 'string',
            format: 'date-time',
        },
        releasedAt: {
            type: 'string',
            format: 'date-time',
        },
        approvedAt: {
            type: 'string',
            format: 'date-time',
        },
        updatedAt: {
            type: 'string',
            format: 'date-time',
        },
        paidAt: {
            type: 'string',
            format: 'date-time',
        },
        cancelledAt: {
            type: 'string',
            format: 'date-time',
        },
        createdBy: {
            type: 'string',
        },
        updatedBy: {
            type: 'string',
        },
        externalId: {
            type: 'string',
        },
        payout: {
            $ref: '#/components/schemas/Payout',
        },
    },
};
export const $PaymentAccount = {
    type: 'object',
    properties: {
        id: {
            type: 'integer',
            format: 'int64',
        },
        userId: {
            type: 'string',
        },
        accountId: {
            type: 'string',
        },
        accountStatus: {
            type: 'string',
        },
        createdAt: {
            type: 'string',
            format: 'date-time',
        },
        connectedAt: {
            type: 'string',
            format: 'date-time',
        },
    },
};
export const $Payout = {
    type: 'object',
    properties: {
        id: {
            type: 'integer',
            format: 'int64',
        },
        paymentAccount: {
            $ref: '#/components/schemas/PaymentAccount',
        },
        createdAt: {
            type: 'string',
            format: 'date-time',
        },
        amount: {
            type: 'integer',
            format: 'int32',
        },
        status: {
            type: 'string',
        },
        transferId: {
            type: 'string',
        },
    },
};
export const $CreatePaymentAccountResponse = {
    type: 'object',
    properties: {
        url: {
            type: 'string',
        },
    },
};
export const $UpdatePaymentRequest = {
    type: 'object',
    properties: {
        id: {
            type: 'integer',
            format: 'int64',
        },
        amount: {
            type: 'integer',
            format: 'int32',
        },
        description: {
            type: 'string',
        },
        approve: {
            type: 'boolean',
        },
    },
};
export const $PaymentDetails = {
    type: 'object',
    properties: {
        id: {
            type: 'integer',
            format: 'int64',
        },
        userId: {
            type: 'string',
        },
        amount: {
            type: 'integer',
            format: 'int32',
        },
        description: {
            type: 'string',
        },
        status: {
            type: 'string',
        },
        externalId: {
            type: 'string',
        },
        originalAmount: {
            type: 'integer',
            format: 'int32',
        },
        approvedAt: {
            type: 'string',
            format: 'date-time',
        },
        paidAt: {
            type: 'string',
            format: 'date-time',
        },
    },
};
export const $PaymentStats = {
    type: 'object',
    properties: {
        draftPaymentsAmount: {
            type: 'integer',
            format: 'int32',
        },
        pendingPaymentsAmount: {
            type: 'integer',
            format: 'int32',
        },
        totalEarnings: {
            type: 'integer',
            format: 'int32',
        },
    },
};
export const $ClientPaymentStats = {
    type: 'object',
    properties: {
        totalExpenses: {
            type: 'integer',
            format: 'int32',
        },
    },
};
