// @republic-global/payment-client/dist/src/services.gen.js
import { OpenAPI } from './core/OpenAPI.js';
import { request as __request } from './core/request.js';
/**
 * List payments by reference ids
 * references are set by external services. for example 'contract-{id}'
 * @param data The data for the request.
 * @param data.referenceIds
 * @returns PaymentDetails OK
 * @throws ApiError
 */
export const getReferencePayments = (data) => {
    return __request(OpenAPI, {
        method: 'GET',
        url: '/payments',
        query: {
            referenceIds: data.referenceIds,
        },
    });
};
/**
 * Create new payment
 * @param data The data for the request.
 * @param data.requestBody
 * @returns Payment OK
 * @throws ApiError
 */
export const createPayment = (data) => {
    return __request(OpenAPI, {
        method: 'POST',
        url: '/payments',
        body: data.requestBody,
        mediaType: 'application/json',
    });
};
/**
 * Update existing payment
 * @param data The data for the request.
 * @param data.requestBody
 * @returns Payment OK
 * @throws ApiError
 */
export const updatePayment = (data) => {
    return __request(OpenAPI, {
        method: 'PATCH',
        url: '/payments',
        body: data.requestBody,
        mediaType: 'application/json',
    });
};
/**
 * Cancel a payment
 * @param data The data for the request.
 * @param data.id
 * @returns Payment OK
 * @throws ApiError
 */
export const cancelPayment = (data) => {
    return __request(OpenAPI, {
        method: 'POST',
        url: '/payments/{id}/cancel',
        path: {
            id: data.id,
        },
    });
};
/**
 * Approve a payment
 * @param data The data for the request.
 * @param data.id
 * @returns Payment OK
 * @throws ApiError
 */
export const approvePayment = (data) => {
    return __request(OpenAPI, {
        method: 'POST',
        url: '/payments/{id}/approve',
        path: {
            id: data.id,
        },
    });
};
/**
 * Get users connected payment account
 * Success -> account connected. Failure -> account not connected
 * @param data The data for the request.
 * @param data.userId
 * @returns PaymentAccount OK
 * @throws ApiError
 */
export const getPaymentAccount = (data) => {
    return __request(OpenAPI, {
        method: 'GET',
        url: '/paymentAccount/{userId}',
        path: {
            userId: data.userId,
        },
        errors: {
            404: 'Payment account not connected',
        },
    });
};
/**
 * Get the url for payment account onboarding. Redirect the user to this url immediately, don't share it anywhere else
 * Success -> do the redirect. Failure -> something is wrong, contact support
 * @param data The data for the request.
 * @param data.userId
 * @returns CreatePaymentAccountResponse OK
 * @throws ApiError
 */
export const createPaymentAccount = (data) => {
    return __request(OpenAPI, {
        method: 'POST',
        url: '/paymentAccount/{userId}',
        path: {
            userId: data.userId,
        },
        errors: {
            500: 'Internal error, contact support',
        },
    });
};
/**
 * List user payments
 * @param data The data for the request.
 * @param data.userId
 * @param data.page
 * @param data.perPage
 * @returns PaymentDetails OK
 * @throws ApiError
 */
export const getUserPayments = (data) => {
    return __request(OpenAPI, {
        method: 'GET',
        url: '/payments/{userId}',
        path: {
            userId: data.userId,
        },
        query: {
            page: data.page,
            perPage: data.perPage,
        },
    });
};
/**
 * Fetch user payment stats
 * @param data The data for the request.
 * @param data.userId
 * @returns PaymentStats OK
 * @throws ApiError
 */
export const getUserPaymentStats = (data) => {
    return __request(OpenAPI, {
        method: 'GET',
        url: '/payments/{userId}/stats',
        path: {
            userId: data.userId,
        },
    });
};
/**
 * Fetch payment stats
 * @returns ClientPaymentStats OK
 * @throws ApiError
 */
export const getPaymentStats = () => {
    return __request(OpenAPI, {
        method: 'GET',
        url: '/payments/stats',
    });
};
/**
 * @returns string OK
 * @throws ApiError
 */
export const all = () => {
    return __request(OpenAPI, {
        method: 'GET',
        url: '/health',
    });
};
