// @republic-global/commons-messaging/dist/src/configurer.js
import { getEnvOrDefault } from '@republic-global/commons';
export const CC_DIRECT_X = 'crowdcraft.direct';
export const crowdcraftQueueTopology = (domain, queueNames) => {
    const queues = queueNames.map((queueName) => {
        return {
            name: `${queueName}`,
            bindings: [
                {
                    exchange: CC_DIRECT_X,
                    routingKey: `${queueName}.key`,
                },
            ],
            options: {
                durable: true,
                deadLetterExchange: CC_DIRECT_X,
                deadLetterRoutingKey: `dead.${domain}.key`,
            },
        };
    });
    const dlq = {
        name: `${domain}.dlq`,
        options: { durable: true },
        bindings: [
            {
                exchange: 'crowdcraft.direct',
                routingKey: `dead.${domain}.key`,
            },
        ],
    };
    return {
        exchanges: [
            {
                name: CC_DIRECT_X,
                type: 'direct',
                options: { durable: true },
            },
        ],
        queues: [...queues, dlq],
    };
};
export const mqConnectionConfig = {
    protocol: getEnvOrDefault('RABBIT_MQ_PROTOCOL', 'amqp'),
    hostname: getEnvOrDefault('RABBIT_MQ_HOST', 'localhost'),
    port: parseInt(getEnvOrDefault('RABBIT_MQ_PORT', '5672')),
    username: getEnvOrDefault('RABBIT_MQ_USERNAME', 'mq'),
    password: getEnvOrDefault('RABBIT_MQ_PASSWORD', 'mq'),
    vhost: getEnvOrDefault('RABBIT_VHOST', '/'),
};
export const mqPublisherChannelOpts = {
    maxChannelCount: parseInt(getEnvOrDefault('RABBIT_MQ_PUBLISHER_MAX_CHANNELS', '2')),
    confirm: getEnvOrDefault('RABBIT_MQ_PUBLISHER_CONFIRM_CHANNEL', 'false') ===
        'true',
};
export const mqSubscriberChannelOpts = {
    maxChannelCount: parseInt(getEnvOrDefault('RABBIT_MQ_SUBSCRIBER_MAX_CHANNELS', '2')),
    confirm: getEnvOrDefault('RABBIT_MQ_SUBSCRIBER_CONFIRM_CHANNEL', 'false') ===
        'true',
};
export const mqConnectionOpts = {
    reconnectTimeInSeconds: parseInt(getEnvOrDefault('RABBIT_MQ_SUBSCRIBER_MAX_CHANNELS', '5')),
    heartbeatIntervalInSeconds: parseInt(getEnvOrDefault('RABBIT_MQ_SUBSCRIBER_MAX_CHANNELS', '30')),
};
