// @republic-global/party-client/dist/src/schemas.gen.js
// This file is auto-generated by @hey-api/openapi-ts
export const $ClientsList = {
    type: 'array',
    items: {
        allOf: [
            {
                $ref: '#/components/schemas/BaseParty',
            },
        ],
    },
};
export const $OrganizationInfo = {
    type: 'object',
    properties: {
        size: {
            type: 'number',
        },
        founded: {
            type: 'string',
        },
        phone: {
            type: 'string',
        },
        category: {
            type: 'string',
        },
    },
};
export const $OrganizationInput = {
    type: 'object',
    properties: {
        name: {
            type: 'string',
        },
        description: {
            type: 'string',
        },
        email: {
            type: 'string',
        },
        address: {
            $ref: '#/components/schemas/AddressInput',
        },
        imageLink: {
            type: 'string',
        },
        socialLinks: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SocialMediaInput',
            },
        },
        size: {
            type: 'number',
        },
        founded: {
            type: 'string',
        },
        phone: {
            type: 'string',
        },
        category: {
            type: 'string',
        },
    },
};
export const $Organization = {
    type: 'object',
    properties: {
        id: {
            type: 'string',
        },
        size: {
            type: 'number',
        },
        founded: {
            type: 'string',
        },
        phone: {
            type: 'string',
        },
        category: {
            type: 'string',
        },
    },
};
export const $PartyFullData = {
    allOf: [
        {
            $ref: '#/components/schemas/BaseParty',
        },
        {
            type: 'object',
            properties: {
                craftsPerson: {
                    $ref: '#/components/schemas/CraftsPersonFullData',
                },
            },
        },
    ],
};
export const $PartyInput = {
    type: 'object',
    properties: {
        name: {
            type: 'string',
        },
        role: {
            type: 'string',
        },
        description: {
            type: 'string',
        },
        email: {
            type: 'string',
        },
        imageLink: {
            type: 'string',
        },
        address: {
            $ref: '#/components/schemas/AddressInput',
        },
        socialLinks: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SocialMediaInput',
            },
        },
    },
};
export const $AddressInput = {
    type: 'object',
    properties: {
        street: {
            type: 'string',
            example: 'Legacy street',
        },
        state: {
            type: 'string',
            example: 'Nevada',
        },
        postalCode: {
            type: 'string',
            example: 12345,
        },
        city: {
            type: 'string',
            example: 'Las Vegas',
        },
        country: {
            type: 'string',
            example: 'UK',
        },
    },
};
export const $SocialMediaInput = {
    type: 'object',
    properties: {
        title: {
            type: 'string',
            example: 'X',
        },
        link: {
            type: 'string',
            example: 'https://twitter.com/@martinfowler',
        },
    },
    required: ['title', 'link'],
};
export const $Address = {
    type: 'object',
    properties: {
        id: {
            type: 'number',
        },
        street: {
            type: 'string',
        },
        city: {
            type: 'string',
        },
        state: {
            type: 'string',
        },
        country: {
            type: 'string',
        },
        postalCode: {
            type: 'string',
        },
        number: {
            type: 'number',
        },
    },
    example: {
        id: '55d2db4d-8824-49cd-b754-cf6c81385d90',
        street: 'Cavara',
        city: 'San Jose',
        state: 'Pensylvania',
        country: 'Santo Domingo',
        postalCode: '1234ABC',
        numer: 13,
    },
};
export const $SocialMedia = {
    type: 'object',
    properties: {
        id: {
            type: 'number',
        },
        title: {
            type: 'string',
        },
        link: {
            type: 'string',
        },
    },
    example: {
        id: 2,
        title: 'LinkedIn',
        link: 'https://linkedin.com/@john123',
    },
};
export const $CraftsPerson = {
    type: 'object',
    properties: {
        craftspersonId: {
            type: 'string',
            example: '8eaf4ae6-cd83-46b0-9534-ea76aad06864',
        },
        resumeLink: {
            type: 'string',
            example: 'https://my.resume.example.com/',
        },
        introduction: {
            type: 'string',
            example: 'I am a skillful craftsperson with x years of experience',
        },
        videoLink: {
            type: 'string',
            example: 'https://my.video.link.example.com/',
        },
        payment: {
            type: 'number',
            example: 200,
        },
        paymentType: {
            type: 'string',
            enum: ['HOURLY', 'FIXED', 'PRIZE'],
        },
    },
};
export const $CraftspersonProfile = {
    allOf: [
        {
            type: 'object',
            properties: {
                ccId: {
                    type: 'string',
                    example: 'c19ba2a3-e9d2-4426-ad0a-33f7bdc8cb5d',
                },
                name: {
                    type: 'string',
                    example: 'Bob',
                },
                surname: {
                    type: 'string',
                    example: 'Martin',
                },
                title: {
                    type: 'string',
                    example: 'King of the world',
                },
                imageLink: {
                    type: 'string',
                    example: 'https://my.image.example.com',
                },
                address: {
                    $ref: '#/components/schemas/Address',
                },
                skills: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/Skill',
                    },
                },
                lastSeen: {
                    type: 'string',
                    example: '2024-06-12T10:16:59.407Z',
                },
            },
        },
        {
            $ref: '#/components/schemas/CraftsPerson',
        },
    ],
};
export const $CraftsPersonFullData = {
    allOf: [
        {
            $ref: '#/components/schemas/CraftsPerson',
        },
        {
            type: 'object',
            properties: {
                skills: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/Skill',
                    },
                },
                education: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/Education',
                    },
                },
                workExperience: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/WorkExperience',
                    },
                },
                portfolio: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/Portfolio',
                    },
                },
            },
        },
    ],
};
export const $CraftsPersonPartyInput = {
    allOf: [
        {
            $ref: '#/components/schemas/PartyInput',
        },
        {
            type: 'object',
            properties: {
                age: {
                    type: 'number',
                },
                resumeLink: {
                    type: 'string',
                },
                videoLink: {
                    type: 'string',
                },
            },
        },
    ],
};
export const $CraftspersonResumeInfo = {
    allOf: [
        {
            type: 'object',
            properties: {
                ccId: {
                    type: 'string',
                    format: 'uuid',
                },
                name: {
                    type: 'string',
                    example: 'Linus',
                },
                surname: {
                    type: 'string',
                    example: 'Trovald',
                },
                birthdate: {
                    type: 'string',
                    format: 'date-time',
                },
                imageUrl: {
                    type: 'string',
                    example: 'https://upload.wikimedia.org/wikipedia/commons/thumb/2/27/Robert_C._Martin_surrounded_by_computers.jpg/762px-Robert_C._Martin_surrounded_by_computers.jpg',
                },
                socialMedia: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/SocialMediaInput',
                    },
                },
                address: {
                    $ref: '#/components/schemas/AddressInput',
                },
                qualification: {
                    $ref: '#/components/schemas/QualificationEnum',
                },
                skills: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/SkillInput',
                    },
                },
            },
        },
        {
            $ref: '#/components/schemas/CraftspersonUpdateInput',
        },
    ],
};
export const $CraftspersonUpdateInput = {
    type: 'object',
    properties: {
        resumeLink: {
            type: 'string',
            example: 'https://my.resume.link.example',
        },
        payment: {
            type: 'number',
            example: 200,
        },
        paymentType: {
            $ref: '#/components/schemas/PaymentTypeEnum',
        },
        introduction: {
            type: 'string',
            example: 'I am a very professional drying paint watcher',
        },
        videoLink: {
            type: 'string',
            example: 'https://my.video.link.example',
        },
        education: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/EducationInput',
            },
        },
        workExperience: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/WorkExperienceInput',
            },
        },
        portfolio: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/PortfolioInput',
            },
        },
    },
    required: [
        'resumeLink',
        'payment',
        'paymentType',
        'introduction',
        'videoLink',
        'education',
        'workExperience',
        'portfolio',
    ],
};
export const $SkillInput = {
    type: 'object',
    properties: {
        name: {
            type: 'string',
            example: 'Dry paint watching',
        },
        proficiency: {
            type: 'number',
            example: 5,
        },
    },
};
export const $WorkExperienceInput = {
    type: 'object',
    properties: {
        title: {
            type: 'string',
            example: 'Lead drying paint watcher',
        },
        company: {
            type: 'string',
            example: 'Microsoft',
        },
        startDate: {
            type: 'string',
            example: '2024-05-30T15:05:10.448Z',
        },
        endDate: {
            type: 'string',
            example: '2024-05-30T15:05:10.448Z',
        },
        description: {
            type: 'string',
            example: 'I spent 10 days as a lead drying paint watcher...',
        },
    },
};
export const $PortfolioInput = {
    type: 'object',
    properties: {
        title: {
            type: 'string',
            example: 'Blue paint dry',
        },
        imageLink: {
            type: 'string',
            example: 'https://blue.dry.paint.example',
        },
        description: {
            type: 'string',
            example: 'I watched this blue painted wall dry for 5 days',
        },
        url: {
            type: 'string',
            example: 'https://blue.dry.paint.project.example',
        },
    },
};
export const $EducationInput = {
    type: 'object',
    properties: {
        title: {
            type: 'string',
            example: 'Master of dry paint',
        },
        academy: {
            type: 'string',
            example: 'Harvard',
        },
        startDate: {
            type: 'string',
            example: '2024-05-30T15:05:10.448Z',
        },
        endDate: {
            type: 'string',
            example: '2024-05-30T15:05:10.448Z',
        },
        description: {
            type: 'string',
            example: 'I spend 4 years at Harvard mastering the art of drying paint watching',
        },
        educationLevel: {
            $ref: '#/components/schemas/QualificationEnum',
        },
    },
};
export const $Client = {
    type: 'object',
    properties: {
        id: {
            type: 'string',
        },
        organization_id: {
            type: 'string',
        },
    },
};
export const $ClientPartyInput = {
    allOf: [
        {
            $ref: '#/components/schemas/PartyInput',
        },
    ],
};
export const $ClientInput = {
    type: 'object',
    properties: {
        organizations: {
            type: 'array',
            items: {
                type: 'string',
            },
        },
    },
};
export const $Education = {
    type: 'object',
    properties: {
        id: {
            type: 'number',
            example: 1,
        },
        title: {
            type: 'string',
            example: 'Master on applied engineering',
        },
        accademy: {
            type: 'string',
            example: 'Greendale',
        },
        dateStart: {
            type: 'string',
            example: '2024-06-12T12:26:47.336Z',
        },
        dateEnd: {
            type: 'string',
            example: '2024-06-12T12:26:47.336Z',
        },
        level: {
            enum: ['ASSOCIATE', 'BACHELOR', 'MASTER', 'DOCTOR'],
            example: 'MASTER',
        },
    },
};
export const $PortfolioItem = {
    type: 'object',
    properties: {
        title: {
            type: 'string',
            example: 'My Awesome project',
        },
        description: {
            type: 'string',
            example: 'I made a clay puppet with blue clay',
        },
        imageLink: {
            type: 'string',
            example: 'https://image.link.example',
        },
        url: {
            type: 'string',
            example: 'https://my.clay.puppet.example',
        },
    },
    required: ['title', 'imageLink', 'url'],
};
export const $Portfolio = {
    type: 'object',
    properties: {
        id: {
            type: 'number',
        },
        title: {
            type: 'string',
        },
        imageLink: {
            type: 'string',
        },
        description: {
            type: 'string',
        },
    },
};
export const $WorkExperience = {
    type: 'object',
    properties: {
        id: {
            type: 'number',
        },
        title: {
            type: 'string',
        },
        company: {
            type: 'string',
        },
        dateStart: {
            type: 'string',
        },
        dateEnd: {
            type: 'string',
        },
    },
};
export const $Skill = {
    type: 'object',
    properties: {
        id: {
            type: 'number',
        },
        name: {
            type: 'string',
        },
        proficiency: {
            type: 'number',
        },
    },
    example: {
        id: '2',
        name: 'Woodpecking',
        proficiency: 5,
    },
};
export const $BaseParty = {
    type: 'object',
    properties: {
        ccId: {
            type: 'string',
            example: 'eb3c671d-a870-4bcc-9f74-8634e2ca30c9',
        },
        name: {
            type: 'string',
            example: 'John',
        },
        surname: {
            type: 'string',
            example: 'Doe',
        },
        birthdate: {
            type: 'string',
            example: '2024-06-10T11:34:23.007Z',
        },
        description: {
            type: 'string',
            example: 'I am a talented drying paint watcher',
        },
        title: {
            type: 'string',
            example: 'Micro Influencer',
        },
        email: {
            type: 'string',
            example: 'john@doe.example',
        },
        countryCode: {
            type: 'string',
            example: 'US',
        },
        phone: {
            type: 'string',
            example: 892324829,
        },
        partyType: {
            type: 'string',
            example: 'Individual',
        },
        imageLink: {
            type: 'string',
            example: 'https://profile.pic.example',
        },
        lastSeen: {
            type: 'string',
            example: '2024-05-30T08:43:03.864Z',
        },
    },
};
export const $CraftspersonWithSkills = {
    type: 'object',
    properties: {
        resumeLink: {
            type: 'string',
        },
        videoLink: {
            type: 'string',
        },
        payment: {
            type: 'number',
        },
        paymentType: {
            type: 'string',
        },
        skills: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SkillInfo',
            },
        },
    },
};
export const $PartyCraftsListItem = {
    allOf: [
        {
            $ref: '#/components/schemas/BaseParty',
        },
        {
            $ref: '#/components/schemas/CraftspersonWithSkills',
        },
        {
            type: 'object',
            properties: {
                address: {
                    $ref: '#/components/schemas/AddressInfo',
                },
                portfolio: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/PortfolioItem',
                    },
                },
            },
        },
    ],
};
export const $CreateParty = {
    type: 'object',
    properties: {
        id: {
            type: 'string',
            format: 'uuid',
        },
        name: {
            type: 'string',
            example: 'Steve',
        },
        surname: {
            type: 'string',
            example: 'Stevenson',
        },
        email: {
            type: 'string',
            example: 'steve@example.com',
        },
        country: {
            type: 'string',
            example: 'Fiji',
        },
    },
    required: ['name', 'id', 'email', 'surname', 'country'],
};
export const $AccountData = {
    type: 'object',
    properties: {
        name: {
            type: 'string',
            example: 'Bob',
        },
        surname: {
            type: 'string',
            example: 'Martin',
        },
        email: {
            type: 'string',
            example: 'unclebob@aol.com',
        },
        phone: {
            type: 'string',
            example: 892324829,
        },
    },
    required: ['name', 'surname', 'email', 'phone'],
};
export const $CraftspersonProfileView = {
    type: 'object',
    allOf: [
        {
            type: 'object',
            properties: {
                ccId: {
                    type: 'string',
                    example: '8a57ca82-c563-448e-a64a-fd203b89bec5',
                },
                name: {
                    type: 'string',
                    example: 'Robert',
                },
                surname: {
                    type: 'string',
                    example: 'Martin',
                },
                imageLink: {
                    type: 'string',
                    example: 'https://my.image.example.com',
                },
                address: {
                    $ref: '#/components/schemas/Address',
                },
                socialMedia: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/SocialMedia',
                    },
                },
                joinDate: {
                    type: 'string',
                },
                qualification: {
                    type: 'string',
                    example: 'Master',
                },
                birthdate: {
                    type: 'string',
                    example: '2024-06-06T11:22:27.765Z',
                },
                education: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/Education',
                    },
                },
                workExperience: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/WorkExperienceInput',
                    },
                },
                skills: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/Skill',
                    },
                },
                portfolio: {
                    type: 'array',
                    items: {
                        $ref: '#/components/schemas/Portfolio',
                    },
                },
            },
        },
        {
            $ref: '#/components/schemas/CraftsPerson',
        },
    ],
};
export const $CraftspersonProfileInfo = {
    type: 'object',
    properties: {
        craftspersonId: {
            type: 'string',
            example: 'fce812f8-c98c-43d5-b5cb-fa77c84b1fd2',
        },
        imageUrl: {
            type: 'string',
            example: 'https://martinfowler.com/img/mf-dallas.jpg',
        },
        name: {
            type: 'string',
            example: 'Martin',
        },
        surname: {
            type: 'string',
            example: 'Fowler',
        },
        title: {
            type: 'string',
            example: 'Software Engineer',
        },
        birthdate: {
            type: 'string',
            example: '2024-06-06T11:22:27.765Z',
        },
        address: {
            $ref: '#/components/schemas/AddressInfo',
        },
        skills: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SkillInput',
            },
        },
        overview: {
            type: 'string',
        },
        socialMedia: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SocialMediaInput',
            },
        },
    },
};
export const $CraftspersonProfileUpdate = {
    type: 'object',
    properties: {
        imageUrl: {
            type: 'string',
            example: 'https://martinfowler.com/img/mf-dallas.jpg',
        },
        title: {
            type: 'string',
            example: 'Software Engineer',
        },
        birthdate: {
            type: 'string',
            example: '2024-06-06T11:22:27.765Z',
        },
        overview: {
            type: 'string',
            example: 'I am one of the best software engineers of all time',
        },
        socialMedia: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SocialMediaInput',
            },
        },
        skills: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SkillInput',
            },
        },
        address: {
            $ref: '#/components/schemas/AddressInput',
        },
    },
    required: [
        'imageUrl',
        'title',
        'birthdate',
        'overview',
        'socialMedia',
        'skills',
        'address',
    ],
};
export const $AddressInfo = {
    type: 'object',
    properties: {
        street: {
            type: 'string',
            example: 'Elm street 13',
        },
        city: {
            type: 'string',
            example: 'Paradise City',
        },
        state: {
            type: 'string',
            example: 'California',
        },
        country: {
            type: 'string',
            example: 'Puerto Rico',
        },
        postalCode: {
            type: 'string',
            example: '12335',
        },
    },
};
export const $SkillInfo = {
    type: 'object',
    properties: {
        name: {
            type: 'string',
            example: 'woodpecking',
        },
        proficiency: {
            type: 'number',
            example: 5,
        },
    },
};
export const $PaymentTypeEnum = {
    type: 'string',
    default: 'HOURLY',
    enum: ['HOURLY', 'FIXED_PRICE', 'PRIZE_POOL'],
};
export const $CraftspersonPartyList = {
    type: 'object',
    properties: {
        records: {
            type: 'array',
            items: {
                allOf: [
                    {
                        $ref: '#/components/schemas/BaseParty',
                    },
                    {
                        type: 'object',
                        properties: {
                            address: {
                                $ref: '#/components/schemas/AddressInfo',
                            },
                            skills: {
                                type: 'array',
                                items: {
                                    $ref: '#/components/schemas/SkillInfo',
                                },
                            },
                            resumeLink: {
                                type: 'string',
                                example: 'https://my-resume.crowdcraft.com',
                            },
                            introduction: {
                                type: 'string',
                                example: "Hi, I am Bob Martin, also known as 'Uncle Bob'",
                            },
                            videoLink: {
                                type: 'string',
                                example: 'https://video-example.crowdcraft.com',
                            },
                            payment: {
                                type: 'number',
                                example: 100,
                            },
                            paymentType: {
                                $ref: '#/components/schemas/PaymentTypeEnum',
                            },
                        },
                    },
                ],
            },
        },
    },
};
export const $CreateCraftsperson = {
    type: 'object',
    properties: {
        id: {
            type: 'string',
            format: 'uuid',
        },
        name: {
            type: 'string',
            example: 'John',
        },
        surname: {
            type: 'string',
            example: 'Carmack',
        },
        email: {
            type: 'string',
            example: 'jcarmack@republic.global',
        },
        country: {
            type: 'string',
            example: 'Cuba',
        },
    },
    required: ['id', 'name', 'surname', 'email', 'country'],
};
export const $CraftspersonPublicProfile = {
    type: 'object',
    properties: {
        ccId: {
            type: 'string',
            format: 'uuid',
        },
        name: {
            type: 'string',
            example: 'Martin',
        },
        surname: {
            type: 'string',
            example: 'Fowler',
        },
        imageLink: {
            type: 'string',
            example: 'https://martin.profile.jpg',
        },
        address: {
            $ref: '#/components/schemas/AddressInfo',
        },
        socialMedia: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SocialMedia',
            },
        },
        joinDate: {
            type: 'string',
            format: 'date-time',
        },
        qualification: {
            $ref: '#/components/schemas/QualificationEnum',
        },
        birthdate: {
            type: 'string',
            format: 'date-time',
        },
        education: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/EducationInfo',
            },
        },
        workExperience: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/WorkExperienceInput',
            },
        },
        skills: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/SkillInfo',
            },
        },
        portfolio: {
            type: 'array',
            items: {
                $ref: '#/components/schemas/PortfolioInput',
            },
        },
        resumeLink: {
            type: 'string',
            example: 'https://crowdcraft.io/my-resume.pdf',
        },
        introduction: {
            type: 'string',
            example: 'My name is Michael Carmack and I am not who you think I am',
        },
        videoLink: {
            type: 'string',
            example: 'https://video.link.example/sda.mp4',
        },
        payment: {
            type: 'number',
            example: 1000,
        },
        paymentType: {
            $ref: '#/components/schemas/PaymentTypeEnum',
        },
    },
};
export const $EducationInfo = {
    type: 'object',
    properties: {
        title: {
            type: 'string',
            example: 'Applied Mathematics',
        },
        academy: {
            type: 'string',
            example: 'Tokyo University',
        },
        dateStart: {
            type: 'string',
            format: 'date-time',
        },
        dateEnd: {
            type: 'string',
            format: 'date-time',
        },
        level: {
            $ref: '#/components/schemas/QualificationEnum',
        },
    },
};
export const $QualificationEnum = {
    type: 'string',
    default: 'BACHELOR',
    enum: ['ASSOCIATE', 'BACHELOR', 'MASTER', 'DOCTOR'],
};
